Dfe.Academies.Academisation.Data/AcademisationContext.cs

```diff
@@ -42,9 +43,21 @@ public override int SaveChanges()
 			{
 				continue;
 			}
-
+			
 			foreach (object? child in children)
 			{
+				Type childType = child.GetType();
```

Comment:
Must make this more generic -- but not in this PR, it's big enough already.

Reply:
yeah, was just trying to get this working @wystanpalm-madetech , definately a hacky solution. Will probably get @paullocknimble to look into this and refactor